tests/src/test/spark400/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala

```diff
@@ -84,6 +89,14 @@ class MockTaskContext(taskAttemptId: Long, partitionId: Int) extends TaskContext
 
   override private[spark] def getLocalProperties = new Properties()
 
+  override private[spark] def interruptible(): Boolean = false
+
+  override private[spark] def pendingInterrupt(
+      threadToInterrupt: Option[Thread], reason: String): Unit = {}
+
+  override private[spark] def createResourceUninterruptibly[T <: Closeable](
+      resourceBuilder: => T): T = resourceBuilder
+
```

Comment:
I wonder if we have to Shim this at all. It looks like the 400 case is just a superset of methods in the pre-400 case without modifying any preexisting methods. The signature don't seem to reference anything that would not compile under pre-400 versions.

Reply:
If this works, you can add a note about such use case to https://github.com/NVIDIA/spark-rapids/blob/branch-25.02/docs/dev/shims.md

Reply:
Thanks @gerashegalov ! Updated the PR as per review comment and added the note in the documentation. PTAL